File: 1-two-sum/1-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        s = {}
        for i in range(len(nums)):
            c = target - nums[i]
            if c in s:
                return [i, s[c]]
            s[nums[i]] = i

File: 1-two-sum/1-two-sum.java
import java.util.HashMap;

class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> hm = new HashMap<>();
        for (int i = 0; i < nums.length; ++i) {
            if (hm.containsKey(target - nums[i])) {
                int[] output = { i, hm.get(target - nums[i]) };
                return output;
            }
            hm.put(nums[i], i);
        }
        return new int[2];
    }
}